### PR TITLE
Prefer using the Promise API

### PR DESCRIPTION
**Pull request by jelly**
For once we ever convert this to a real Promise API

### CODE REVIEW

File: src/app.jsx [hunk 1]
```diff
@@ -513,7 +513,7 @@ class Application extends React.Component {
                         });
                     }
                 })
-                .fail(e => console.log("Could not read $XDG_RUNTIME_DIR: ", e.message));
+                .catch(e => console.log("Could not read $XDG_RUNTIME_DIR: ", e.message));
```
Comment (cockpituous) [inline src/app.jsx:516]: This added line is not executed by any test.

**Review by martinpitt — APPROVED** [review]

**Review by martinpitt — CHANGES_REQUESTED** [review]
Wait, while you are at it, can you please convert this, too?
```
src/app.jsx:                .done(xrd => {
```

**Comment by jelly** [comment]
> Wait, while you are at it, can you please convert this, too?
> 
> ```
> src/app.jsx:                .done(xrd => {
> ```

What do you mean?

**Comment by martinpitt** [comment]
I mean this: https://github.com/cockpit-project/cockpit-podman/blob/cfe4a9cc845b63fd912297f2af615ec354b7276c/src/app.jsx#L501

`.done()` is jquery API, please change to `.then()`

**Review by martinpitt — APPROVED** [review]

**Pull request merged**